### PR TITLE
CIAC-9828 HashiCorp Vault Audit Logs Modeling

### PR DESCRIPTION
**Pull request by cweltPA**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [CIAC-9828](https://jira-dc.paloaltonetworks.com/browse/CIAC-9828)

## Description
This PR adds Cortex XSIAM SIEM content for ingesting and normalizing HashiCorp Vault Audit logs to the [XDM schema](https://docs-cortex.paloaltonetworks.com/r/Cortex-XSIAM/Cortex-Data-Model-Schema-Guide/Introduction). The content in this PR include: 

- Parsing Rules.
- Modeling Rules.
- XDRC Filebeat configuration Templates for Windows & Linux platforms. 

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/HashiCorp-Vault/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<~XSIAM>
+
+## What does this pack do?
+
+In addition to the [HasiCorp Vault Integration](https://xsoar.pan.dev/docs/reference/integrations/hashi-corp-vault) for managing secrets and credentials on HashiCorp Vault, this pack includes *Cortex Data Modeling (XDM) Rules* and *Parsing Rules* for ingesting and normalizing [HashiCorp Vault Audit Logs](https://support.hashicorp.com/hc/en-us/articles/360001722947-Audit-Device-Notes). 
+
+## Configuration 
+Follow the steps bellow for configuring ingestion of Hashicorp Vault audit logs into Cortex XSIAM.
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorp-Vault/README.md:8]:
```suggestion
Follow the steps below to configure ingestion of Hashicorp Vault audit logs into Cortex XSIAM.
```

File: Packs/HashiCorp-Vault/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+$ vault audit enable file file_path=/var/log/vault_audit.log
+```
+See also: 
+- [Log file rotation](https://developer.hashicorp.com/vault/docs/audit/file#log-file-rotation).
+- [Audit Devices](https://developer.hashicorp.com/vault/docs/audit).
+- [Blocked audit devices](https://developer.hashicorp.com/vault/tutorials/monitoring/blocked-audit-devices).
+
+### Configuration on Cortex XSIAM 
+  
+1. Install the HashiCorp Vault content pack from the Cortex XSIAM marketplace.
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorp-Vault/README.md:23]:
```suggestion
1. Install the HashiCorp Vault content pack from Cortex XSIAM Marketplace. 
```

File: Packs/HashiCorp-Vault/README.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+- [Audit Devices](https://developer.hashicorp.com/vault/docs/audit).
+- [Blocked audit devices](https://developer.hashicorp.com/vault/tutorials/monitoring/blocked-audit-devices).
+
+### Configuration on Cortex XSIAM 
+  
+1. Install the HashiCorp Vault content pack from the Cortex XSIAM marketplace. 
+2. Configure an [XDR Collector](https://docs-cortex.paloaltonetworks.com/r/Cortex-XSIAM/Cortex-XSIAM-Documentation/Manage-XDR-Collectors):
+   1. Create an XDR Collector installation package as described [here](https://docs-cortex.paloaltonetworks.com/r/Cortex-XSIAM/Cortex-XSIAM-Documentation/Create-an-XDR-Collector-installation-package).
+   2. Install the XDR Collector created installation package on the HashiCorp Vault server: 
+      - For a *Windows* server see [Install the XDR Collector installation package for Windows](https://docs-cortex.paloaltonetworks.com/r/KkeZwTYbDACMoWxJk0COqg/g4A0gTe8mLEmRNXk9hhsbw).
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorp-Vault/README.md:27]:
```suggestion
      - For a *Windows* server see [Install the XDR Collector installation package for Windows](https://docs-cortex.paloaltonetworks.com/r/Cortex-XSIAM/Cortex-XSIAM-Documentation/Install-the-XDR-Collector-installation-package-for-Windows).
```

File: Packs/HashiCorp-Vault/README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      - For a *Linux* server see [Install the XDR Collector installation package for Linux](https://docs-cortex.paloaltonetworks.com/r/KkeZwTYbDACMoWxJk0COqg/upKreUev0g1LzaCpxHJIRA).
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorp-Vault/README.md:28]:
```suggestion
      - For a *Linux* server see [Install the XDR Collector installation package for Linux](https://docs-cortex.paloaltonetworks.com/r/Cortex-XSIAM/Cortex-XSIAM-Documentation/Install-the-XDR-Collector-installation-package-for-Linux). 
```

File: Packs/HashiCorp-Vault/ReleaseNotes/1_1_24.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+##### New: HashiCorp Vault Parsing Rule
+
+<~XSIAM> Added [Parsing Rules](https://docs-cortex.paloaltonetworks.com/r/Cortex-XSIAM/Cortex-XSIAM-Administrator-Guide/Parsing-Rules) for ingesting [Hashicorp Vault Audit Logs](https://support.hashicorp.com/hc/en-us/articles/360001722947-Audit-Device-Notes) into the *`hashicorp_vault_raw`* dataset and populating the system *`_time`* field with the raw audit log *`time`* field timestamp value (Available from Cortex XSIAM 2.3). </~XSIAM>
+
+#### XDRC Templates
+
+##### New: HashiCorp Vault (Linux)
+<~XSIAM>  
+- New: Added an XDR Collector [Filebeat](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-filestream.html) profile template for ingesting audit logs from a HashiCorp Vault *Linux* server via a [file audit device](https://developer.hashicorp.com/vault/docs/audit/file) (Available from Cortex XSIAM 2.3).</~XSIAM>
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorp-Vault/ReleaseNotes/1_1_24.md:23]:
```suggestion
New: Added an XDR Collector [Filebeat](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-filestream.html) profile template for ingesting audit logs from a HashiCorp Vault *Linux* server via a [file audit device](https://developer.hashicorp.com/vault/docs/audit/file) (Available from Cortex XSIAM 2.3).</~XSIAM>
```

File: Packs/HashiCorp-Vault/ReleaseNotes/1_1_24.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+##### New: HashiCorp Vault (Windows)
+<~XSIAM>  
+- New: Added an XDR Collector [Filebeat](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-filestream.html) profile template for ingesting audit logs from a HashiCorp Vault *Windows* server via a [file audit device](https://developer.hashicorp.com/vault/docs/audit/file) (Available from Cortex XSIAM 2.3).</~XSIAM>
```
Comment (ShirleyDenkberg) [inline Packs/HashiCorp-Vault/ReleaseNotes/1_1_24.md:26]:
```suggestion
New: Added an XDR Collector [Filebeat](https://www.elastic.co/guide/en/beats/filebeat/current/filebeat-input-filestream.html) profile template for ingesting audit logs from a HashiCorp Vault *Windows* server via a [file audit device](https://developer.hashicorp.com/vault/docs/audit/file) (Available from Cortex XSIAM 2.3).</~XSIAM>
```

**Comment by ShirleyDenkberg** [comment]
@OBavly Doc review completed.

**Review by OBavly — APPROVED** [review]

**Pull request merged**